src/core/cgroup.c

```diff
@@ -3627,7 +3627,7 @@ int unit_search_main_pid(Unit *u, PidRef *ret) {
         for (;;) {
                 _cleanup_(pidref_done) PidRef npidref = PIDREF_NULL;
 
-                r = cg_read_pidref(f, &npidref);
+                r = cg_read_pidref(f, &npidref, CGROUP_DONT_SKIP_UNMAPPED);
```

Comment:
Why 0 is critical here??

Reply:
It's to have read_pidref() return an error when an unmappable pid is encountered, since it's not considered sane to have them appear in a unit.

Reply:
This was asked by me in https://github.com/systemd/systemd/pull/32722#discussion_r1595238115

Reply:
OK. Then could you add a short comment about that?

Reply:
Sure, done.